### PR TITLE
HDDS-5844. Provide capability to config ratis grpcLogAppender max pending request and log write buffer size

### PR DESCRIPTION
**Pull request by ChenSammi**
https://issues.apache.org/jira/browse/HDDS-5844

### CODE REVIEW

**Comment by bharatviswa504** [comment]
@ChenSammi In Datanode we can add any ratis GRPC config by appending with hdds.ratis.raft.grpc.<<ratis property>> and hdds.ratis.raft.server.<<ratisconfig>>. The idea behind this is we donot need to we can add any ratis config with out any code changes.

`we'm tring to find a set of recommended ratis property configurations for a new high density HDD`
For this we can have doc update in our documentation which we feel recommended and explain about them instead of adding to code, so that end users can know about them, than adding to these Java classes. Thoughts??

**Comment by ChenSammi** [comment]
> 
> 
> @ChenSammi In Datanode we can add any ratis GRPC config by appending with hdds.ratis.raft.grpc.`ratis config` and hdds.ratis.raft.server.`ratisconfig`. The idea behind this is we donot need to we can add any ratis config with out any code changes.

I see. This is a really good feature. 

> 
> `we'm tring to find a set of recommended ratis property configurations for a new high density HDD` For this we can have doc update in our documentation which we feel recommended and explain about them instead of adding to code, so that end users can know about them, than adding to these Java classes. Thoughts??

Yes, we are feeling the same way.  Ratis has many configuratins, some of them are actually related. We'd like to find the rule about this. For example, given the DN disk number, available memory,  NIC bandwidth, and the expected read/write concurrency of the whole cluster,  what's the best configration set.

**Comment by adoroszlai** [comment]
/pending skip adding new properties (prefer prefix-based approach)

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/XceiverServerRatis.java [hunk 1]
```diff
@@ -413,7 +415,10 @@ private void setRaftSegmentAndWriteBufferSize(RaftProperties properties) {
     RaftServerConfigKeys.Log.setSegmentSizeMax(properties,
         SizeInBytes.valueOf(raftSegmentSize));
     RaftServerConfigKeys.Log.setWriteBufferSize(properties,
-            SizeInBytes.valueOf(raftSegmentSize));
+        SizeInBytes.valueOf(conf.getObject(DatanodeRatisServerConfig.class)
```
[the remark this answers is not in the document]
Comment (adoroszlai) [reply hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/XceiverServerRatis.java:418]: @bharatviswa504 @bshashikant can you please answer @ChenSammi's question about max. segment size vs. write buffer size?

**Pull request closed without merging**